src/app/core/login/login.component.html

```diff
@@ -1,3 +1,11 @@
-<p>
-  login works!
-</p>
+<mat-card>
+  <form [formGroup]="form" (ngSubmit)="login()">
```

Comment:
where is mat-card selector declared? are you using any custom tool or something?

Reply:
Yup...I'm using Angular Material - https://material.angular.io/components/card/overview

Reply:
https://dzone.com/articles/implementing-guard-in-angular-5-app I can see that entire code is similar as this one.

Reply:
Also, auth guards can be implemented by using observables alone, will you try that. Won't be approving copy-paste code. Thanks

Reply:
Hmm.. okay... I will try more then

Reply:
I'm closing this PR and opening a new one.